### PR TITLE
Refactor HookEvent to avoid DI mistake and clean LegacyHookSubscriber

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Refactor HookEvent to avoid DI mistake and clean LegacyHookSubscriber
| Type?         | refacto
| Category?     | CO
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | No need for QA ?

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19429)

### CODE REVIEW

File: src/PrestaShopBundle/Service/Hook/HookEvent.php [hunk 1]
```diff
@@ -54,25 +81,12 @@ public function setHookParameters($parameters)
...
-     * These values can either be overriden by setHookParameters using the same parameter key.
-     *
-     * @return array the array of hook parameters, more default fixed values
+     * @return array
      */
     public function getHookParameters()
     {
-        $globalParameters = ['_ps_version' => AppKernel::VERSION];
-
-        $sfContainer = SymfonyContainer::getInstance();
```
Comment (matks) [inline src/PrestaShopBundle/Service/Hook/HookEvent.php:69]: This is the horror Dependency Injection I wanted to clean

Comment (Progi1984) [reply src/PrestaShopBundle/Service/Hook/HookEvent.php:69]: Good job ;)

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
@matks But Travis is :red_circle:

**Comment by matks** [comment]
Technically I introduce a BC break because `(new HookEvent())->getHookParameters();` return has changed

**Comment by matks** [comment]
I broke CI
```
* Installing PrestaShop, this may take a while ...
Call to a member function get() on nullInstalled failed, displaying errors from logs:
```

**Comment by matks** [comment]
Fixed install

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @matks

**Pull request merged**